### PR TITLE
RM#18375 INVOICE : generate pdf file for supplier invoice

### PR DESCRIPTION
**Pull request by jdo-axelor**

### CODE REVIEW

File: axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java [hunk 1]
```diff
@@ -854,4 +857,30 @@ public Boolean checkPartnerBankDetailsList(Invoice invoice) {
 
     return true;
   }
+  
+  protected boolean checkEnablePDFGenerationOnVentilation(Invoice invoice) 
+  {
+	  if(appAccountService.getAppInvoice().getIsEnablePDFGenerationForCustomersOnVentilation()) 
+	  {
```
Comment (gdu-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java]: Better to use the helper to know if it's a sale or purchase invoice/refund :
https://github.com/axelor/axelor-business-suite/blob/master/axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceToolService.java#L89

Comment (jdo-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java]: It's done, check if it's correct.

File: axelor-account/src/main/resources/views/AppInvoice.xml [hunk 2]
```diff
@@ -9,6 +9,8 @@
             <field name="isVentilationSkipped" widget="boolean-switch"/>
 			<field name="isEnabledProductDescriptionCopyForCustomers" widget="boolean-switch"/>
 			<field name="isEnabledProductDescriptionCopyForSuppliers" widget="boolean-switch"/>
+			<field name="isEnablePDFGenerationForCustomersOnVentilation" widget="boolean-switch"/>
```
Comment (gdu-axelor) [inline axelor-account/src/main/resources/views/AppInvoice.xml]: name of these parameters should be reviewed : 
autoGenerateInvoicePrintingFileOnPurchaseInvoice
autoGenerateInvoicePrintingFileOnSaleInvoice

File: axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java [hunk 3]
```diff
@@ -854,4 +857,19 @@ public Boolean checkPartnerBankDetailsList(Invoice invoice) {
 
     return true;
   }
+  
+  protected boolean checkEnablePDFGenerationOnVentilation(Invoice invoice) throws AxelorException 
+  {
+	  	//isPurchase() = isSupplier()
+		if(appAccountService.getAppInvoice().getIsEnablePDFGenerationForCustomersOnVentilation() && !InvoiceToolService.isPurchase(invoice)) 
+		{
```
Comment (gdu-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java]: Indentation is not good. Please apply google code format.

File: axelor-account/src/main/resources/domains/AppInvoice.xml [hunk 4]
```diff
@@ -10,12 +10,16 @@
 		<boolean name="isVentilationSkipped" title="Skip ventilation"/>
 		<boolean name="isEnabledProductDescriptionCopyForCustomers" title="Enable product description copy for customers"/>
 		<boolean name="isEnabledProductDescriptionCopyForSuppliers" title="Enable product description copy for suppliers"/>
+		<boolean name="isEnablePDFGenerationForCustomersOnVentilation" title="Enable PDF generation for customers on ventilation"/>
```
Comment (gdu-axelor) [inline axelor-account/src/main/resources/domains/AppInvoice.xml]: default value should be TRUE

File: axelor-account/src/main/resources/domains/AppInvoice.xml [hunk 5]
```diff
@@ -10,12 +10,16 @@
...
+		<boolean name="isEnablePDFGenerationForSuppliersOnVentilation" title="Enable PDF generation for suppliers on ventilation"/>
```
Comment (gdu-axelor) [inline axelor-account/src/main/resources/domains/AppInvoice.xml]: default value should be FALSE

File: axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java (hunk 3 above)
Comment (jdo-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java]: Done

File: axelor-account/src/main/resources/domains/AppInvoice.xml (hunk 4 above)
Comment (jdo-axelor) [reply axelor-account/src/main/resources/domains/AppInvoice.xml]: Done

File: axelor-account/src/main/resources/domains/AppInvoice.xml (hunk 5 above)
Comment (jdo-axelor) [reply axelor-account/src/main/resources/domains/AppInvoice.xml]: Done

**Pull request merged**